File: BOJ/5568. 카드 놓기/solution.js
const input = require('fs').readFileSync('/dev/stdin').toString().trim().split('\n');
const k = +input[1];
const isVisited = Array(input[0]).fill(0);
const nums = input.slice(2);

const set = new Set();
let temp = [];
function dfs(depth) {
    if (depth === k) {
        set.add(temp.join(''));
        return;
    }
    
    for (let i = 0; i < nums.length; i += 1) {
        if (!isVisited[i]) {
            isVisited[i] = 1;
            temp.push(nums[i]);
            dfs(depth + 1);
            temp.pop();
            isVisited[i] = 0;
        }
    }
}
dfs(0);
console.log(set.size);

File: BOJ/5568. 카드 놓기/solution.cc
#include <iostream>
#include <vector>
#include <unordered_set>

using namespace std;

int n, k, nums[11], visited[11];
unordered_set<string> s;
vector<int> temp;

void dfs(int depth) {
    if (depth == k) {
        string t;
        for (int x: temp) t += to_string(x);
        s.insert(t);
        return;
    }

    for (int i = 0; i < n; i += 1) {
        if (!visited[i]) {
            visited[i] = 1;
            temp.push_back(nums[i]);
            dfs(depth + 1);
            temp.pop_back();
            visited[i] = 0;
        }
    }
}

int main() {
    ios_base::sync_with_stdio(false);
    cin.tie(0);

    cin >> n >> k;

    for (int i = 0; i < n; i += 1) {
        cin >> nums[i];
    }
    dfs(0);
    cout << s.size();
    return 0;
}